dev-docs/services/ci/README.md

```diff
@@ -0,0 +1,244 @@
+of the `ci` billing project. For example: `/batches?q=user+%3D+ci%0D%0Atest+%3D+1`.
+
+#### CI Testing Timeline
+
+The image below shows the CI testing timeline:
+
+```mermaid
+sequenceDiagram
+    participant Github
+    box Kubernetes (PROD: 'default' namespace)
+    participant CI as PROD CI Service (default namespace)
+    participant PB as PROD Batch (default namespace)
+    end
+    box GCE (PROD VMs)
+    participant PVM as  PROD VMs
+    end
+    box Kubernetes (CI: 'test-xyz' namespaces)
+    participant CIB as CI Batch (test-xyz namespaces)
+    end
+    box GCE (CI VMs)
+    participant CIVM as CI VMs
+    end
+
+    Github->>CI: Notify PR created/updated
+
+    CI->>Github: Register github check (pending)
+    CI->>PB: Submit CI test batch
+    activate PB
+    PB->>PVM: Submit build jobs
+    activate PVM
+    PVM-->>PB: Done
+    deactivate PVM
+    PB->>PVM: Submit deploy jobs
+    activate PVM
+    PVM->>CIB: Deploy batch service
+    activate CIB
+    PVM-->>PB: Done
+    deactivate PVM
+    PB->>PVM: Submit test jobs
+    activate PVM
+    PVM->>CIB: Submit test batches
+    CIB->>CIVM: Submit test batch jobs
+    activate CIVM
+    CIVM-->>CIB: Validate results
+    deactivate CIVM
+    PVM-->>PB: Done
+    deactivate PVM
+    PB->>PVM: Submit cleanup jobs
+    activate PVM
+    PVM->>CIB: Clean up service
+    deactivate CIB
+    PVM-->>PB: Done
+    deactivate PVM
+    PB-->>CI: Detects completion
+    CI->>Github: Update github check
+```
+
+## Merging PRs to the `main` branch
+
+When a PR state changes, it will cause the `github_changed` flag to become dirty. 
+
+During its update loop, the CI service will determine whether any PRs targeting its watched branches are elibible to
+be merged.
+
+Readiness is determined by github status checks. The following conditions must be met:
+
+- The PR must be against the `main` branch
+- The PR must have passed all tests in GCP
+- The PR must be approved
+
+The control flow from final approval to CI merging a PRs looks like:
+
+- The PR's state will change in github (either a check changes to SUCCESS, or a review is approved)
+- The github webhook callback will cause the `github_changed` flag will be marked as dirty for the `WatchedBranch`
+- The `WatchedBranch`'s `_update` method in [`github.py`](../../../ci/ci/github.py) scans all PRs against the branch and updates state that helps determine mergeability.
+- The `WatchedBranch`'s `_update` method in [`github.py`](../../../ci/ci/github.py) iterates again to merge all mergeable PRs, in priority order
+
+## Deploying services to the live infrastructure
+
+When a PR is merged into the `main` branch, a webhook will trigger. The CI service will set its `github_changed` flag.
+
+During its update loop, the CI service will determine that the SHA of the `WatchedBranch` has changed and trigger
+a deployment.
+
+- Create a new batch job to:
```

Comment:
I believe this is also `build.yaml`, just with a configuration variable set to make it deploy?